src/runners/scilla_runner.ml

```diff
@@ -115,6 +115,36 @@ let rec output_event_json elist =
     | _ -> `Null :: j)
   | [] -> []
 
+let validate_get_init_json init_file gas_remaining sourcever =
+  (* Retrieve initial parameters *)
+  let initargs =
+    try
+      JSON.ContractState.get_json_data init_file
+    with
+    | Invalid_json s ->
+        fatal_error_gas (s @ (mk_error0 (sprintf "Failed to parse json %s:\n" init_file))) gas_remaining
+  in
+  (* Check for version mismatch. Subtract penalty for mismatch. *)
+  let emsg, rgas = (mk_error0 ("Scilla version mismatch\n")),
+    (Uint64.sub gas_remaining (Uint64.of_int Gas.version_mismatch_penalty))
+  in
+  let init_json_scilla_version = List.fold_left initargs ~init:None ~f:(fun found (name, lit) ->
+    if is_some found then found else
+    if name = ContractUtil.scilla_version_label
+    then match lit with | UintLit(Uint32L v) -> Some v | _ -> None
+    else None
```

Comment:
This function does two things at once and it's a bit hard to read. I'd suggest using a `List.Assoc.find` and then extract the found Scilla version as a separate step. And perhaps throw an exception if there is something else than a Uint32 literal associated to `scilla_version_label`.

Reply:
ok done.

Reply:
Awesome! Thanks a lot!